components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/prometheus/reporter/PrometheusReporter.java

```diff
@@ -0,0 +1,266 @@
+                TOTAL_REQUESTS_RECEIVED_INBOUND_ENDPOINT = Counter.build
+                        (MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_TOTAL, metricHelp).
+                        labelNames(labels).register();
+                metricMap.put(metricName, TOTAL_REQUESTS_RECEIVED_INBOUND_ENDPOINT);
+
+            } else if (type.equals(MetricConstants.HISTOGRAM)) {
+                INBOUND_ENDPOINT_LATENCY_HISTOGRAM = Histogram.build()
+                        .name(MetricConstants.INBOUND_ENDPOINT_LATENCY_SECONDS)
+                        .help(metricHelp)
+                        .labelNames(labels)
+                        .buckets(inboundEndpointLatencyBuckets)
+                        .register();
+                metricMap.put(metricName, INBOUND_ENDPOINT_LATENCY_HISTOGRAM);
+            }
+        } else if (serviceType.equals(MetricConstants.SERVER)) {
+            SERVER_UP = Gauge.build(MetricConstants.SERVER_UP, "Server status").
+                    labelNames(labels).register();
+            metricMap.put(MetricConstants.SERVER_UP, SERVER_UP);
+
+        } else {
+            SERVICE_UP = Gauge.build(MetricConstants.SERVICE_UP, "Service status").
+                    labelNames(labels).register();
+            metricMap.put(MetricConstants.SERVICE_UP, SERVICE_UP);
+        }
+    }
+
+    @Override
+    public void initErrorMetrics(String serviceType, String type, String metricName, String metricHelp, Map<String,
+            String[]> properties) {
+
+        String[] labels = properties.get(metricName);
+
+        if (serviceType.equals(SERVICE.PROXY.name())) {
+            ERROR_REQUESTS_RECEIVED_PROXY_SERVICE = Counter.build(MetricConstants.PROXY_REQUEST_COUNT_ERROR_TOTAL,
+                    metricHelp).
+                    labelNames(labels).register();
+            metricMap.put(metricName, ERROR_REQUESTS_RECEIVED_PROXY_SERVICE);
+        } else if (serviceType.equals(SERVICE.API.name())) {
+            ERROR_REQUESTS_RECEIVED_API = Counter.build(MetricConstants.API_REQUEST_COUNT_ERROR_TOTAL, metricHelp).
+                    labelNames(labels).register();
+            metricMap.put(metricName, ERROR_REQUESTS_RECEIVED_API);
+        } else if (serviceType.equals(SERVICE.INBOUND_ENDPOINT.name())) {
+            ERROR_REQUESTS_RECEIVED_INBOUND_ENDPOINT = Counter.
+                    build(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_ERROR_TOTAL, metricHelp).labelNames(labels).
+                    register();
+            metricMap.put(metricName, ERROR_REQUESTS_RECEIVED_INBOUND_ENDPOINT);
+
+        }
+    }
+
+    @Override
+    public void incrementCount(String metricName, Map<String, String[]> properties) {
+        Counter counter = (Counter) metricMap.get(metricName);
+        String[] value = properties.get(metricName);
+
+        counter.labels(value).inc();
+    }
+
+    @Override
+    public void decrementCount(String metricName, Map<String, String> properties) {
+
```

Comment:
Any particular reason to leave it empty?

Reply:
decrementCount() is not necessary to be implemented for the Prometheus Reporter
 as Gauge is used in Prometheus for the metrics that can both increment and decrement value.
Added a comment explaining the reason.